nilearn/glm/tests/test_regression.py

```diff
@@ -1,45 +1,49 @@
 """Test functions for models.regression"""
 
-import numpy as np
+import pytest
 from numpy.testing import assert_almost_equal, assert_array_almost_equal
 
 from nilearn.glm import ARModel, OLSModel
 
-RNG = np.random.RandomState(42)
-X = RNG.standard_normal(size=(40, 10))
-Y = RNG.standard_normal(size=(40,))
 
+@pytest.fixture()
+def X(rng):
+    return rng.standard_normal(size=(40, 10))
 
-def test_OLS():
+
+@pytest.fixture()
+def Y(rng):
+    return rng.standard_normal(size=(40, 10))
+
+
+def test_OLS(X, Y):
     model = OLSModel(design=X)
     results = model.fit(Y)
     assert results.df_residuals == 30
     assert results.residuals.shape[0] == 40
     assert results.predicted.shape[0] == 40
 
 
-def test_AR():
+def test_AR(X, Y):
     model = ARModel(design=X, rho=0.4)
     results = model.fit(Y)
     assert results.df_residuals == 30
     assert results.residuals.shape[0] == 40
     assert results.predicted.shape[0] == 40
 
 
-def test_residuals():
-    Xintercept = X.copy()
-
+def test_residuals(X, Y):
     # If design matrix contains an intercept, the
     # mean of the residuals should be 0 (short of
     # some numerical rounding errors)
-    Xintercept[:, 0] = 1
-    model = OLSModel(design=Xintercept)
+    X[:, 0] = 1
```

Comment:
Isn't X modified ?

Reply:
yeah but it gets regenerated every time the fixture is invoked so eventhough it is modified in this test, the next test will get a "fresh new X".